configure

```diff
@@ -211,8 +205,8 @@ check_define __APPLE__ && {
 	mac_detected=true
 	check_define __x86_64__ && mac_64=true
 	if test "$hookmethod" = auto ; then
-		osver=$(get_define_stripped __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ 2>/dev/null)
-		test "$osver" -gt $((120000 - 1)) && hookmethod=dyld
+		osver=`sw_vers | grep -i productver | awk '{print $2}' | awk -F. '{print $1}'`
```

Comment:
this is not compatible with cross-compiling (you cannot call sw_vers on a linux device compiling for OSX), therefore we need to find a better way to fix this.

Reply:
Maybe we can keep both `sw_vers` and `__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__`?
That is to say, the larger version wins.

Reply:
@rofl0r Pls take another look at the newest commit.